### PR TITLE
vx stream integration

### PR DESCRIPTION
**Pull request by asafshen**
fixes:  https://github.com/demisto/server/issues/8231

### CODE REVIEW

File: Integrations/integration-VxStream.yml [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+
+        // create context from response
+        var context = {};
+
+        var contextMapper = {
+            sha1: 'SHA1',
+            sha256: 'SHA256',
+            md5: 'MD5',
+        };
+        context[outputPaths.file] = response.map(mapObject(contextMapper));
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: what about adding malicious?

File: Integrations/integration-VxStream.yml [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                var newKey = mapper[key] || key;
+                res[newKey] = obj[key];
+            });
+            return res;
+        }
+    }
+
+    function createTableEntry(name, rawResponse, table, context, headers) {
+        return {
+            Type: entryTypes.note,
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: remove comments

File: Integrations/integration-VxStream.yml [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+        };
+    }
+
+    var sendRequest = function(uri) {
+        var requestUrl = serverUrl + uri
+
+        var res = http(
+            requestUrl,
+            {
+                Method: 'GET', // Can be GET, POST, PUT, DELETE, HEAD, OPTIONS or CONNECT
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: remove comments

**Comment by asafshen** [comment]
@Spivaka fixed.

Image is still missing.. working on it with Ido

**Comment by asafshen** [comment]
@Spivaka whenever possible

**Comment by dorsha** [comment]
@asafshen what is the status of this pr?

**Comment by asafshen** [comment]
Waiting for review

File: Integrations/integration-VxStream.yml [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+configuration:
+- display: Server URL (e.g. https://216.3.128.82)
+  name: serverUrl
+  defaultvalue: https://www.hybrid-analysis.com
+  type: 0
+  required: true
+- display: API Key
+  name: apiKey
+  defaultvalue: ""
+  type: 0
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: should be of type encrypted

File: Integrations/integration-VxStream.yml [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+  required: true
+- display: Secret Key
+  name: secretKey
+  defaultvalue: ""
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: seems that this also should be encrypted

File: Integrations/integration-VxStream.yml [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+           if(res.threatlevel && res.threatlevel > 1) {
+               addMalicious(context, outputPaths.file, {
+                    MD5: res.md5,
+                    SHA1: res.sha1,
+                    SHA256: res.sha256,
+                    Malicious: { Vendor: 'VXStream', Description: 'Score above ' + res.threatscore }
+                });
+           }
+        });
+        return createTableEntry('Scan Results:', response, table, context);
```
Comment (Spivaka) [inline Integrations/integration-VxStream.yml]: might be better to try catch or anyway other to handle non-json response

Comment (asafshen) [reply Integrations/integration-VxStream.yml]: handled errors above it after `sendRequest` according to status code

Comment (Spivaka) [reply Integrations/integration-VxStream.yml]: yes, in some cases the response is 200 but the body is not in json format (some proxies for example)

Comment (asafshen) [reply Integrations/integration-VxStream.yml]: In this specific integration?
At all integrations?

Comment (Spivaka) [reply Integrations/integration-VxStream.yml]: at all. seen it with vectra and others

**Comment by gshriki** [comment]
@asafshen - when are you planning to merge this?

**Comment by asafshen** [comment]
@gshriki Merged

**Comment by gshriki** [comment]
Thanks!

**Pull request merged**